includes/blocks/course-list/class-sensei-course-list-block.php

```diff
@@ -73,4 +74,53 @@ public function maybe_change_inherited_to_true( $parsed_block ) {
 		}
 		return $parsed_block;
 	}
+
+	/**
+	 * Add an attribute in inner blocks inside post-template block to determine if they're in Query Loop block.
+	 *
+	 * @param array  $parsed_block The block to be rendered.
+	 * @param array  $source_block Unchanged source copy of the block to be rendered.
+	 * @param object $parent_block The parent block.
+	 *
+	 * @access private
+	 * @return array
+	 */
+	public function maybe_add_attributes_to_inner_blocks( $parsed_block, $source_block, $parent_block ) {
+		if (
+			'core/post-template' === $parsed_block['blockName'] &&
+			$parent_block &&
+			'core/query' === ( $parent_block->parsed_block['blockName'] ?? '' ) &&
+			'course' === ( $parent_block->parsed_block['attrs']['query']['postType'] ?? '' )
+		) {
+			$parsed_block['innerBlocks'] = $this->recursively_set_attribute_to_inner_blocks( $parsed_block['innerBlocks'] );
```

Comment:
I haven't reviewed the associated Pro PR yet, but is it necessary to add this attribute to all the inner blocks and not just the Take Course one?

Reply:
> is it necessary to add this attribute to all the inner blocks instead of just the Take Course one?

Right, we just need it in the Take Course one. I just added it to all inner blocks just so that all inner blocks know they are inside of a Course List block, but it seems that's not necessary. I've removed it here to target only the Take Course block https://github.com/Automattic/sensei/pull/6457/commits/486fc4f5f79423ee293bcb20ffa1198674025b92